dash/development/_r_components_generation.py

```diff
@@ -625,3 +641,74 @@ def generate_exports(
         package_imports,
         package_suggests,
     )
+
+
+def get_r_prop_types(type_object):
+    """Mapping from the PropTypes js type object to the R type"""
+
+    return dict(
+        array=lambda: "unnamed list",
+        bool=lambda: "logical",
+        number=lambda: "numeric",
+        string=lambda: "character",
+        object=lambda: "named list",
+        any=lambda: 'logical | numeric | character | named list | unnamed list',  # noqa:E501
+        element=lambda: "dash component",
+        node=lambda: "a list of or a singular dash " "component, string or number",  # noqa:E501
+        # React's PropTypes.oneOf
+        enum=lambda: "a value equal to: {}".format(
+            ", ".join("{}".format(str(t["value"]))
+                      for t in type_object["value"])
+        ),
+        # React's PropTypes.oneOfType
+        union=lambda: "{}".format(
+            " | ".join(
+                "{}".format(get_r_type(subType))
+                for subType in type_object["value"]
+                if get_r_type(subType) != ""
+            )
+        ),
+        # React's PropTypes.arrayOf
+        arrayOf=lambda: "list".format(  # pylint: disable=too-many-format-args
```

Comment:
I see that's in the python version too but it doesn't look workable. Perhaps it was meant to be
`"list" + (" of {}s".format(...) if... else "")`

Reply:
fixed in 083ff1b